### ISSUE TITLE
groupby + apply gets 'KeyError' when a column is referenced in the UDF 

### ISSUE THREAD

**Issue by eavidan** [modin-project/modin#432]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS 10.14.2
- **Modin installed from (source or binary)**: source
- **Modin version**: 0.3.0
- **Python version**: 3.6.7
- **Exact command to reproduce**: 

### Describe the problem
When doing a `groupby` and then applying a UDF with reference to a column (eg: `df['B']`), we receive a `KeyError: 'B'`. 

### Source code / logs
```
# works
import pandas as pd

df = pd.DataFrame({'A': [2, 2, 2, 3, 4],
                   'B': [5, 6, 7, 8, 9],
                   })

df = df.groupby('A').apply(lambda x: x['B']+1)
print(df)

# does not work
import modin.pandas as pd

df = pd.DataFrame({'A': [2, 2, 2, 3, 4],
                   'B': [5, 6, 7, 8, 9],
                   })

df = df.groupby('A').apply(lambda x: x['B']+1)
print(df)

```

**Comment by devin-petersohn** [comment]
Thanks for report this @eavidan!

We have a different internal indexing structure than the external structure to avoid copying the index/columns for each partition. For apply, it looks like we may just have to copy the columns internally as well.

We will also need to catch `KeyErrors` internally to make sure that we aren't throwing erroneous errors.

**Comment by mvashishtha** [comment]
This is fixed as of Modin version ef654c414364ca319812114324ad9d85e83ee68e.
